spec/system/admin/orders/invoices_spec.rb

```diff
@@ -104,6 +106,67 @@
     end
   end
 
+  describe 'printing invoices' do
+    context 'when the order has no invoices' do
+      it 'creates an invoice for the order' do
+        expect(order.invoices.count).to eq 0
+        page.find("#links-dropdown", text: "ACTIONS").click
+        click_link "Print Invoice"
+        new_window = windows.last
+        page.within_window new_window do
+          expect(order.invoices.count).to eq 1
+        end
```

Comment:
TIL about handling windows with capybara!

However I don't see why we need to here, this is a backend expectation.
Oh... is this to cause a 'wait' until the new window is opened? If so, can we make that clearer with an expectation on the page content? (expect new_window to have_content "invoice" or something like that?)

Reply:
Yes, I needed to force Capybara to wait for the new tab to be loaded. 
Otherwise, it'll reach the assertion before the backend generates the invoice.

Reply:
the new page doesn't have a title and it's not possible to run a search on it as it only contains a PDF.
What's the best approach to fix that? Save the PDF, convert it to text, and check its content?

Reply:
Tricky.. that's a good idea, but sounds fiddly.
For now, I would suggest we just add a comment to explain.

```suggestion
        # wait for PDF to open in new window
        new_window = windows.last
        page.within_window new_window do
          expect(order.invoices.count).to eq 1
        end
```